File: $_algotsu-kaizen/problemo-set/maximum-matching-of-players-with-trainers.c
int cmp(const void *a, const void *b) {
    return (*(int*)b - *(int*)a);
}

int matchPlayersAndTrainers(int* players, int playersSize, int* trainers, int trainersSize) {
    qsort(players, playersSize, sizeof(int), cmp);
    qsort(trainers, trainersSize, sizeof(int), cmp);

    // for (int _ = 0; _ < playersSize; ++_) printf(" %d ", players[_]);
    // printf("\n");
    // for (int _ = 0; _ < trainersSize; ++_) printf(" %d ", trainers[_]);
    register int pidx = 0, tidx = 0, result = 0;

    while (pidx < playersSize && tidx < trainersSize) {
        if (players[pidx] <= trainers[tidx]) {
            ++result;
            ++tidx;
        }

        ++pidx;
    }

    return result;
}

File: $_algotsu-kaizen/problemo-set/maximum-matching-of-players-with-trainers.rs
impl Solution {
    pub fn match_players_and_trainers(mut players: Vec<i32>, mut trainers: Vec<i32>) -> i32 {
        players.sort_by(|a, b| b.cmp(a));
        trainers.sort_by(|a, b| b.cmp(a));
        let (mut pidx, mut tidx, mut result) = (0x0, 0x0, 0x0);

        while pidx < players.len() && tidx < trainers.len() {
            if players[pidx] <= trainers[tidx] {
                result += 0x1;
                tidx += 0x1;
            }

            pidx += 0x1;
        }

        result
    }
}
